manim/mobject/svg/tex_mobject.py

```diff
@@ -113,6 +113,37 @@ def construct(self):
             tex.set_color_by_tex('igsta', RED)
             self.add(tex)
 
+Note that :func:`~.set_color_by_tex` colors the entire substring containing the Tex searched for,
+not just the specific symbol or Tex expression searched for. Consider the following example:
+
+.. manim:: IncorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color_by_tex(YELLOW)
+            self.add(equation)
```

Comment:
I still feel that the incorrect example is too complicated. Why not simply: 

```suggestion
            equation.set_color_by_tex("x", YELLOW)
            self.add(equation)
```

(That's at least what I would initially expect to work.)